app/components/Views/confirmations/components/ApproveTransactionReview/index.js

```diff
@@ -1164,18 +1170,22 @@ class ApproveTransactionReview extends PureComponent {
     } catch (error) {
       Logger.error(error, 'Navigation: Error when navigating to buy ETH.');
     }
-    InteractionManager.runAfterInteractions(() => {
-      Analytics.trackEvent(MetaMetricsEvents.RECEIVE_OPTIONS_PAYMENT_REQUEST);
-    });
+
+    this.props.metrics.trackEvent(
```

Comment:
Destructure metrics on L1165

```suggestion
    metrics.trackEvent(
```